### PR TITLE
Added R-gsignal recipe

### PR DESCRIPTION
**Pull request by dipterix**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [ ] If static libraries are linked in, the license of the static library is packaged.
- [x] Package does not ship static libraries. If static libraries are needed, [follow CFEP-18](https://github.com/conda-forge/cfep/blob/main/cfep-18.md).
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by dipterix** [comment]
@conda-forge/help-r Could you help me with the `missing_dso_whitelist` issue here?

```
ERROR (r-gsignal,lib/R/library/gsignal/libs/x64/gsignal.dll): $RPATH/R.dll not found in packages, sysroot(s) nor the missing_dso_whitelist.
```

https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=1018035&view=logs&j=240f1fee-52bc-5498-a14a-8361bde76ba0&t=7c0f8eae-6d6f-51bf-636f-73a1a7fb1bc4&l=1915

**Comment by mfansler** [comment]
Note that the latest `conda_r_skeleton_helper` (what is [recommended for CRAN packages](https://conda-forge.org/docs/maintainer/adding_pkgs/#generating-the-recipe)) includes a DSO whitelist.

**Comment by dipterix** [comment]
@conda-forge/help-r, ready for review!

File: recipes/r-gsignal/build.sh [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+mv DESCRIPTION DESCRIPTION.old
+grep -va '^Priority: ' DESCRIPTION.old > DESCRIPTION
+# shellcheck disable=SC2086
+${R} CMD INSTALL --build . ${R_ARGS}
+
+# Add more build steps here, if they are necessary.
+
+# See
+# https://docs.conda.io/projects/conda-build
+# for a list of environment variables that are set during the build process.
```
Comment (mfansler) [inline recipes/r-gsignal/build.sh:22]:
```suggestion
```

File: recipes/r-gsignal/build.sh [hunk 2]
```diff
@@ -0,0 +1,22 @@
+#!/bin/bash
+
+# 'Autobrew' is being used by more and more packages these days
+# to grab static libraries from Homebrew bottles. These bottles
+# are fetched via Homebrew's --force-bottle option which grabs
+# a bottle for the build machine which may not be macOS 10.9.
+# Also, we want to use conda packages (and shared libraries) for
+# these 'system' dependencies. See:
+# https://github.com/jeroen/autobrew/issues/3
```
Comment (mfansler) [inline recipes/r-gsignal/build.sh:9]:
```suggestion
```

File: recipes/r-gsignal/build.sh [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+export DISABLE_AUTOBREW=1
+
+# R refuses to build packages that mark themselves as Priority: Recommended
+mv DESCRIPTION DESCRIPTION.old
+grep -va '^Priority: ' DESCRIPTION.old > DESCRIPTION
```
Comment (mfansler) [inline recipes/r-gsignal/build.sh:14]: This is old boilerplate - safe to remove:

```suggestion
```

File: recipes/r-gsignal/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,86 @@
+{% set version = '0.3-5' %}
+{% set posix = 'm2-' if win else '' %}
+
+package:
+  name: r-gsignal
+  version: {{ version|replace("-", "_") }}
+
+source:
+  # curl -sL https://cran.r-project.org/src/contrib/gsignal_0.3-5.tar.gz | openssl sha256
```
Comment (mfansler) [inline recipes/r-gsignal/meta.yaml:9]:
```suggestion
```

File: recipes/r-gsignal/meta.yaml [hunk 5]
```diff
@@ -0,0 +1,86 @@
...
+  url:
+    - {{ cran_mirror }}/src/contrib/gsignal_{{ version }}.tar.gz
+    - {{ cran_mirror }}/src/contrib/Archive/gsignal/gsignal_{{ version }}.tar.gz
+  sha256: 4ebf509626ad7f7ed7e67269958cf135c1de22a703acf8a6e94e9fb7d145cb63
+
+build:
+  merge_build_host: True  # [win]
+  # If this is a new build for the same version, increment the build number.
```
Comment (mfansler) [inline recipes/r-gsignal/meta.yaml:17]:
```suggestion
```

File: recipes/r-gsignal/meta.yaml [hunk 6]
```diff
@@ -0,0 +1,86 @@
...
+  number: 0
+  # no skip
+
+  # This is required to make R link correctly on Linux.
```
Comment (mfansler) [inline recipes/r-gsignal/meta.yaml:21]:
```suggestion
```

File: recipes/r-gsignal/meta.yaml [hunk 7]
```diff
@@ -0,0 +1,86 @@
...
+  rpaths:
+    - lib/R/lib/
+    - lib/
+
+  # DSO whitelist on windows, could be caused by importing other base-R packages
```
Comment (mfansler) [inline recipes/r-gsignal/meta.yaml:26]:
```suggestion
```

File: recipes/r-gsignal/meta.yaml [hunk 8]
```diff
@@ -0,0 +1,86 @@
...
+  missing_dso_whitelist:
+    - '*/R.dll'        # [win]
+    - '*/Rblas.dll'    # [win]
+    - '*/Rlapack.dll'  # [win]
+
+# Suggests: knitr, rmarkdown, testthat, ggplot2, gridExtra, microbenchmark, imager, covr
```
Comment (mfansler) [inline recipes/r-gsignal/meta.yaml:32]:
```suggestion
```

File: recipes/r-gsignal/meta.yaml [hunk 9]
```diff
@@ -0,0 +1,86 @@
...
+    - {{ stdlib("m2w64_c") }}          # [win]
+    - {{ compiler('cxx') }}            # [not win]
+    - {{ compiler('m2w64_cxx') }}      # [win]
+    - cross-r-base {{ r_base }}    # [build_platform != target_platform]
+    - {{ posix }}filesystem        # [win]
+    - {{ posix }}sed               # [win]
+    - {{ posix }}make
+    - {{ posix }}coreutils         # [win]
+    - {{ posix }}zip               # [win]
+
```
Comment (mfansler) [inline recipes/r-gsignal/meta.yaml:47]:
```suggestion
```

File: recipes/r-gsignal/meta.yaml [hunk 10]
```diff
@@ -0,0 +1,86 @@
...
+  host:
+    - r-base
+    - r-rcpp
+    - r-pracma
+
```
Comment (mfansler) [inline recipes/r-gsignal/meta.yaml:52]:
```suggestion
```

File: recipes/r-gsignal/meta.yaml [hunk 11]
```diff
@@ -0,0 +1,86 @@
...
+  run:
+    - r-base
+    - r-rcpp
+    - r-pracma
+
+test:
+  commands:
+    # You can put additional test commands to be run here.
```
Comment (mfansler) [inline recipes/r-gsignal/meta.yaml:60]:
```suggestion
```

**Review by mfansler — CHANGES_REQUESTED** [review]
Thanks for submitting this! ❤️ 

Mostly just some style changes (we prefer not to commit all the skeleton comments), plus a license adjustment.

In the future, please use `conda_r_skeleton_helper` for CRAN packages.

**Comment by dipterix** [comment]
@mfansler I commit all the changes that you have suggested. Sorry I didn't notice that each commit will trigger github action check and is non-stoppable.... Now it's all over the place : O

**Review by mfansler — APPROVED** [review]
Thanks for submitting! ❤️

**Comment by gjmvanboxtel** [comment]
I confirm that I would like to be added as the maintainer

**Pull request merged**